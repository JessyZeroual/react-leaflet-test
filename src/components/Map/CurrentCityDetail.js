import React from "react";
import {
  StyledWrapperCurrentCityDetail,
  StyledCurrentCityDetailText,
} from "./Map.styled";

const CurrentCityDetail = ({ currentCity }) => (
  <>
    <StyledWrapperCurrentCityDetail>
      <img src="/icon/buildings.svg" width="30" alt="icon city" />
      <StyledCurrentCityDetailText>
        {currentCity.city}
      </StyledCurrentCityDetailText>
    </StyledWrapperCurrentCityDetail>
    <StyledWrapperCurrentCityDetail>
      <img src="/icon/flag.svg" width="30" alt="icon state" />
      <StyledCurrentCityDetailText>
        {currentCity.state}
      </StyledCurrentCityDetailText>
    </StyledWrapperCurrentCityDetail>
    <StyledWrapperCurrentCityDetail>
      <img src="/icon/growth.svg" width="30" alt="icon growth" />

      <StyledCurrentCityDetailText
        isNegative={Math.sign(currentCity.growth_from_2000_to_2013) === -1}
      >
        {`${currentCity.growth_from_2000_to_2013}%`}
      </StyledCurrentCityDetailText>
    </StyledWrapperCurrentCityDetail>
    <StyledWrapperCurrentCityDetail>
      <img src="/icon/population.svg" width="30" alt="icon population" />
      <StyledCurrentCityDetailText>
        {currentCity.population.toLocaleString("en-US")}
      </StyledCurrentCityDetailText>
    </StyledWrapperCurrentCityDetail>
  </>
);

export default CurrentCityDetail;
