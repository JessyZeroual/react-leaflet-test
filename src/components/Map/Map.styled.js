import styled from "styled-components";

export const StyledWrapperMap = styled.div`
  border: 1px solid grey;
  margin-left: 20px;
  flex: 1;
`;

export const StyledWrapperCurrentCityDetail = styled.div`
  display: flex;
  align-items: center;
`;

export const StyledCurrentCityDetailText = styled.p`
  font-weight: bold;
  font-size: 15px;
  padding: 0px 10px;
  color: ${({ isNegative }) => isNegative && "red"};
`;
