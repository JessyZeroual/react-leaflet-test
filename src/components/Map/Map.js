import React, { useState, useEffect } from "react";
import { Map as LeafletMap, Marker, Popup, TileLayer } from "react-leaflet";
import { StyledWrapperMap } from "./Map.styled";
import CurrentCityDetail from "./CurrentCityDetail";

const Map = ({ currentCity }) => {
  const [zoom, setZoom] = useState(7);
  const [latitude, setLatitude] = useState(40.7127837);
  const [longitude, setLongitude] = useState(-74.0059413);

  useEffect(() => {
    const animation = () => {
      setLatitude(currentCity.latitude);
      setLongitude(currentCity.longitude);
      setTimeout(() => {
        setZoom(7);
      }, 800);
    };
    setZoom(4);
    if (currentCity.latitude) animation();
  }, [currentCity]);

  return (
    <StyledWrapperMap>
      <LeafletMap center={[latitude, longitude]} zoom={zoom} minZoom={2}>
        <TileLayer url="http://{s}.tile.osm.org/{z}/{x}/{y}.png" />
        <Marker position={[latitude, longitude]}>
          <Popup closeButton={false}>
            <CurrentCityDetail currentCity={currentCity} />
          </Popup>
        </Marker>
      </LeafletMap>
    </StyledWrapperMap>
  );
};

export default Map;
