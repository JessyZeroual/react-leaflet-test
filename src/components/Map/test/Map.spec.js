import React from "react";
import TestRenderer from "react-test-renderer";
import Map from "../Map";
jest.mock("leaflet");

describe("Map", () => {
  let renderer;

  beforeEach(() => {
    renderer = TestRenderer.create(
      <Map
        currentCity={{
          city: "New York",
          growth_from_2000_to_2013: 4.8,
          latitude: 40.7127837,
          longitude: -74.0059413,
          population: 8405837,
          rank: 1,
          state: "New York",
        }}
      />
    );
  });

  describe("When Map is mounted", () => {
    it("renders correctly", () => {
      expect(renderer).toMatchSnapshot();
    });

    it("receives correct properties", () => {
      const { currentCity } = renderer.root.findByType(Map).props;

      expect(typeof currentCity.city).toEqual("string");
      expect(typeof currentCity.growth_from_2000_to_2013).toEqual("number");
      expect(typeof currentCity.latitude).toEqual("number");
      expect(typeof currentCity.longitude).toEqual("number");
      expect(typeof currentCity.population).toEqual("number");
      expect(typeof currentCity.rank).toEqual("number");
      expect(typeof currentCity.state).toEqual("string");
    });
  });
});
