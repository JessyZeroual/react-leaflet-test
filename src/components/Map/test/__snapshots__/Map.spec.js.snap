// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map When Map is mounted renders correctly 1`] = `
<div
  className="sc-AxjAm jykgfV"
>
  <div />
</div>
`;
